src/Gaffer/Animation.cpp

```diff
@@ -245,22 +306,15 @@ bool Animation::CurvePlug::hasKey( float time ) const
 
 Animation::Key *Animation::CurvePlug::getKey( float time )
 {
-	auto it = m_keys.find( time );
-	if( it != m_keys.end() )
-	{
-		return it->get();
-	}
-	return nullptr;
+	return const_cast< Key* >( static_cast< const CurvePlug* >( this )->getKey( time ) );
 }
 
 const Animation::Key *Animation::CurvePlug::getKey( float time ) const
 {
-	auto it = m_keys.find( time );
-	if( it != m_keys.end() )
-	{
-		return it->get();
-	}
-	return nullptr;
+	Keys::const_iterator it = m_keys.find( time );
+	return ( it != m_keys.end() )
+		? &( *it )
+		: static_cast< const Key* >( 0 );
```

Comment:
Is there a benefit of `static_cast<const Key *>( 0 )` over `nullptr`?

Reply:
not really, I'm just used to writing it this way as the type of the expressions in both branches of the ``` ? : ``` have to be the same.

Reply:
done in latest commit